### PR TITLE
variable: define StatementContext.

### PR DESCRIPTION
**Pull request by coocood**
StatementContext is used to hold statement execution mode and state.
As every computation need this, there are many API changes to make, this is just a start.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

File: tidb.go [hunk 1]
```diff
@@ -111,6 +112,24 @@ func Parse(ctx context.Context, src string) ([]ast.StmtNode, error) {
...
 }
 
+// Before every execution, we must clear statement status.
+func resetStmtStatus(ctx context.Context, s ast.StmtNode) {
+	sessVars := ctx.GetSessionVars()
+	var sc variable.StatementContext
+	switch stmt := s.(type) {
+	case *ast.ShowStmt:
+		if stmt.Tp == ast.ShowWarnings {
+			sc.SetWarnings(sessVars.StmtCtx.GetWarnings())
```
Comment (coocood) [inline tidb.go]: No, session variable is created by `NewSessionVars`, in which `StmtCtx` is always created.

**Pull request merged**